src/app/core/permissions/permission-enforcer.service.ts

```diff
@@ -0,0 +1,104 @@
+import { Inject, Injectable } from "@angular/core";
+import { DatabaseRule } from "./permission-types";
+import { SessionService } from "../session/session-service/session.service";
+import { EntityConstructor } from "../entity/model/entity";
+import { DynamicEntityService } from "../entity/dynamic-entity.service";
+import { EntityMapperService } from "../entity/entity-mapper.service";
+import { Database } from "../database/database";
+import { LOCATION_TOKEN } from "../../utils/di-tokens";
+import { AnalyticsService } from "../analytics/analytics.service";
+import { EntityAbility } from "./entity-ability";
+import { User } from "../user/user";
+import { ProgressDashboardConfig } from "../../child-dev-project/progress-dashboard-widget/progress-dashboard/progress-dashboard-config";
+import { Config } from "../config/config";
+
+@Injectable()
+export class PermissionEnforcerService {
+  static readonly STORAGE_KEY = "RULES";
+  private readonly ignoredSubject = [
+    User.ENTITY_TYPE,
+    ProgressDashboardConfig.ENTITY_TYPE,
+    Config.ENTITY_TYPE,
+  ];
```

Comment:
This is because of the default rules in the backend which I added so we dont have to add the necessary rules every time (like updating user doc or fetching config object which are mostly necessary). This could however also be removed and left to the person defining the rules.

Reply:
Move this into a default permission template